include/sqlite_orm/sqlite_orm.h

```diff
@@ -3446,7 +3446,26 @@ namespace sqlite_orm {
 namespace sqlite_orm {
     
     namespace internal {
-        
+
+/*
+ * This is because of bug in MSVC, for more information, please visit
+ * https://stackoverflow.com/questions/34672441/stdis-base-of-for-template-classes/34672753#34672753
+ */
+#if defined(_MSC_VER)
+       template <template <typename...> class Base, typename Derived>
+       struct is_base_of_template_impl {
+               template<typename... Ts>
+               static constexpr std::true_type test(const Base<Ts...>*);
+
+               static constexpr std::false_type test(...);
+
+               using type = decltype(test(std::declval<Derived*>()));
+       };
+
+       template <typename Derived, template <typename...> class Base>
+       using is_base_of_template = typename is_base_of_template_impl<Base, Derived>::type;
```

Comment:
I added preprocessor guard to separate MSVC compatible solution with general and clean solution.

I think it is good to explicitly hold these two code simultaneously, because it is more readable and when MSVC resolved this bug we can remove all bunch of code.

but if you want to remove it, I can remove codes from `#else`  to end of `#endif` and hold MSVC code instead of normal code. but I don't suggest it.

what is your opinion?

Reply:
@fnc12  Are you agree with me?

Reply:
Sorry, my bad. The code is ok. I though that this code is included in unit test instead of dev header. I agree with you